File: practica1/Ejercicio2/Bicicleta.py
from abc import ABC, abstractmethod

class Bicicleta(ABC):
    idBicicleta = none

    def __init__(self, id):
        self.idBicicleta = id

    @abstractmethod
    def clone(self):
        pass

    def correr(self):
        print("Bicicleta " + self.idBicicleta + " corriendo")
        
    def anuncioRetirada(self):
        print("Bicicleta " + self.idBicicleta + " se retira.")

File: practica1/Ejercicio1/src/ejercicio1/Bicicleta.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Interface.java to edit this template
 */
package ejercicio1;

/**
 *
 * @author carmenxufdz
 */
public abstract class Bicicleta {
    protected int idBicicleta;
    
    Bicicleta(int id)
    {
        idBicicleta = id;
    }

    public void correr(){
        System.out.println("Bicicleta " + idBicicleta + " corriendo.");    
    }

    public void anuncioRetirada(){
        System.out.println("Bicicleta " + idBicicleta + " se retira.");
    }
}
